src/libdredd/src/mutate_visitor.cc

```diff
@@ -859,4 +878,33 @@ bool MutateVisitor::IsLvalueCallThatUsesMaterializedTemporary(
   return false;
 }
 
+void MutateVisitor::SaveConstantTemplateArgumentForRewrite(
+    const clang::TemplateSpecializationType* template_specialization_type) {
+  for (auto template_argument :
+       template_specialization_type->template_arguments()) {
+    switch (template_argument.getKind()) {
+      case clang::TemplateArgument::Expression:
+        // Template Argument is Expression
+        if (const auto* expr = template_argument.getAsExpr()) {
```

Comment:
Is it possible for getKind() to return Expression but for getAsExpr() to then fail? Likely not, therefore I would get rid of the if and do something like:

```
const auto* expr = ...;
assert(expr != nullptr && "It should be possible to get a template argument with Expression kind as an Expr");
constant_arguments_to_rewritte.push_back(expr);
```

Reply:
Is it okay to replace with
```
constant_arguments_to_rewritte.push_back(template_argument.getAsExpr());
```
instead, without the assertion checking?

Reply:
Yes, please go ahead. Once done, if bots pass, feel free to merge.